lib/pcb/svg-object-fns/create-svg-objects-from-pcb-copper-text.ts

```diff
@@ -0,0 +1,351 @@
+import type { PcbCopperText } from "circuit-json"
+import type { INode as SvgObject } from "svgson"
+import {
+  applyToPoint,
+  compose,
+  rotate,
+  translate,
+  scale,
+  toString as matrixToString,
+} from "transformation-matrix"
+import type { PcbContext } from "../convert-circuit-json-to-pcb-svg"
+import { layerNameToColor } from "../layer-name-to-color"
+import { distance } from "circuit-json"
+import { svgAlphabet } from "@tscircuit/alphabet"
+
+// Character spacing constants
+const CHAR_WIDTH = 1.0
+const CHAR_SPACING = 0.2
+const LINE_HEIGHT = 1.4
+
+type AlphabetKey = keyof typeof svgAlphabet
+
+/**
+ * Transform a normalized SVG path (0–1 range) to actual coordinates.
+ */
+function transformPathData(
+  pathData: string,
+  offsetX: number,
+  offsetY: number,
+  charScale: number,
+): string {
+  return pathData.replace(
+    /([ML])\s*([-\d.]+)\s+([-\d.]+)|([Q])\s*([-\d.]+)\s+([-\d.]+)\s+([-\d.]+)\s+([-\d.]+)/g,
+    (match, cmd1, x1, y1, cmd2, qx1, qy1, qx2, qy2) => {
+      if (cmd1) {
+        const x = offsetX + Number.parseFloat(x1) * charScale
+        const y = offsetY + Number.parseFloat(y1) * charScale
+        return `${cmd1}${x} ${y}`
+      }
+      if (cmd2) {
+        const cx = offsetX + Number.parseFloat(qx1) * charScale
+        const cy = offsetY + Number.parseFloat(qy1) * charScale
+        const x = offsetX + Number.parseFloat(qx2) * charScale
+        const y = offsetY + Number.parseFloat(qy2) * charScale
+        return `${cmd2}${cx} ${cy} ${x} ${y}`
+      }
+      return match
+    },
+  )
+}
```

Comment:
needlessly complicated, check the README of @tscircuit/alphabet, it exports lines so that you don't need to parse SVGs